server/app/services/AlertSettings.java

```diff
@@ -23,26 +23,59 @@ public record AlertSettings(
     Boolean isSlim) {
 
   public static AlertSettings empty() {
-    return new AlertSettings(false, Optional.empty(), "", AlertType.NONE);
+    return AlertSettings.builder().build();
   }
 
-  public AlertSettings(Boolean show, Optional<String> title, String text, AlertType alertType) {
-    this(show, title, text, alertType, ImmutableList.of(), /* isSlim= */ false);
+  public static AlertSettingsBuilder builder() {
+    return new AlertSettingsBuilder();
   }
 
-  public AlertSettings(
-      Boolean show,
-      Optional<String> title,
-      String text,
-      AlertType alertType,
-      ImmutableList<String> additionalText,
-      Boolean isSlim) {
-    this(show, title, text, /* unescapedDescription= */ true, alertType, additionalText, isSlim);
-  }
+  public static class AlertSettingsBuilder {
+    private Boolean show = false;
+    private Optional<String> title = Optional.empty();
+    private String text = "";
+    private Boolean unescapedDescription = false;
+    private AlertType alertType = AlertType.NONE;
+    private ImmutableList<String> additionalText = ImmutableList.of();
+    private Boolean isSlim = false;
+
+    public AlertSettingsBuilder show(Boolean show) {
+      this.show = show;
+      return this;
+    }
+
+    public AlertSettingsBuilder title(Optional<String> title) {
+      this.title = title;
+      return this;
+    }
+
+    public AlertSettingsBuilder text(String text) {
+      this.text = text;
+      return this;
+    }
+
+    public AlertSettingsBuilder unescapedDescription(Boolean unescapedDescription) {
+      this.unescapedDescription = unescapedDescription;
+      return this;
+    }
+
+    public AlertSettingsBuilder alertType(AlertType alertType) {
+      this.alertType = alertType;
+      return this;
+    }
+
+    public AlertSettingsBuilder additionalText(ImmutableList<String> additionalText) {
+      this.additionalText = additionalText;
+      return this;
+    }
+
+    public AlertSettingsBuilder isSlim(Boolean isSlim) {
+      this.isSlim = isSlim;
+      return this;
+    }
 
-  /** Sanitized HTML for the alert text that processes Markdown. */
-  public String getFormattedAlertText(String text) {
```

Comment:
suggestion (blocking): Keep this method. It is used in the template here: https://github.com/civiform/civiform/blob/3ffef58e5c896e2912b4c6fbc0476b4af6968fa8/server/app/views/components/AlertFragment.html#L39